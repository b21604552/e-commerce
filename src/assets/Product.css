.products-wrapper {
    margin-right: 80px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: 1072px;
    margin-top: 19px;
    margin-bottom: 16px;
}

.product-wrapper {
    width: 254px;
    margin-right: 14px;
    margin-bottom: 16px;
}

.product-wrapper:hover {
    border: 1px solid #E5E5E5;
    border-radius: 10px;
    margin-right: 12px;
}

.product-image-wrapper {
    width: 254px;
    height: 339px;
    background: #FFFFFF;
    border-radius: 10px;
    border: 1px solid #E5E5E5;
    display: flex;
    justify-content: center;
    align-items: center;
}

.product-wrapper:hover > .product-image-wrapper {
    border: 0;
}

.product-info-wrapper {
    margin: 16px 16px 0;
}

.product-wrapper:hover > .product-info-wrapper {
    display: none;
}

.product-title {
    margin-bottom: 12px;
    width: 230px;
    height: 32px;
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 18px;
    color: #484848;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
}

.product-properties-info-wrapper {
    margin-bottom: 12px;
}

.product-properties-element {
    display: flex;
}

.product-properties-label {
    font-style: normal;
    font-weight: 700;
    font-size: 12px;
    line-height: 18px;
    color: #484848;
    margin-right: 5px;
}

.product-properties-value {
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    color: #484848;
}

.product-price-info {

}

.product-price-element {
    font-style: normal;
    font-weight: 700;
    font-size: 12px;
    line-height: 18px;
    color: #000000;
}

.product-old-price-element {
    display: inline-flex;
}

.product-old-price {
    font-style: normal;
    font-weight: normal;
    font-size: 13px;
    line-height: 18px;
    text-decoration-line: line-through;
    color: #9B9B9B;
    margin-right: 3px;
}

.product-discount {
    font-style: normal;
    font-weight: bold;
    font-size: 12px;
    line-height: 18px;
    color: #F90000;
}

.product-wrapper:hover > .product-add-cart-wrapper {
    display: flex;
}

.product-add-cart-wrapper {
    margin-bottom: 20px;
    margin-left: 11px;
    margin-right: 10px;
    display: none;
    justify-content: center;
    align-items: end;
    height: 125px;
    width: 222px;
}

.product-button {
    width: 233px;
    height: 32px;
    background: rgba(255, 96, 0, 0.11);
    border: 0;
    border-radius: 8px;
    color: #FF6000;
    cursor: pointer;
}

.product-button:disabled {
    background: rgba(126, 126, 126, 0.11);
    color: #7E7E7E;
    cursor: not-allowed;
}