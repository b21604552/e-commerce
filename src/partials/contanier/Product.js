import React, { useEffect, useState } from 'react'
import GeneralHelper from '../../helper/GeneralHelper'
import Api from '../../services/Api'
import { useSelector } from 'react-redux'

export default function Product (props) {
  const { itemCountInBasket } = useSelector((state) => state?.GeneralReducer)
  const [data, setData] = useState(props.data)
  const [basket, setBasket] = useState(Api.getBasket)

  useEffect(() => {
    setData(props.data)
  }, [props.data])

  function addProductToBasket (id) {
    Api.addProductToBasket(id)
    setBasket(Api.getBasket())
  }

  useEffect(() => {
    setBasket(Api.getBasket())
  }, [itemCountInBasket])

  return (
    <div className={'products-wrapper'}>
      {data !== null && (
        data.map((product) => (
          <div key={product.id} className="product-wrapper">
            <div className={'product-image-wrapper'}>
              <img alt={'product-' + product.id} src={GeneralHelper.getImageByKey(product.image)}/>
            </div>
            <div className={'product-info-wrapper'}>
              <div className={'product-title'}>
                {product.title}
              </div>
              <div className={'product-properties-info-wrapper'}>
                <div className={'product-properties-element'}>
                  <div className={'product-properties-label'}>
                    Marka:
                  </div>
                  <div className={'product-properties-value'}>
                    {product.brand}
                  </div>
                </div>
                <div className={'product-properties-element'}>
                  <div className={'product-properties-label'}>
                    Renk:
                  </div>
                  <div className={'product-properties-value'}>
                    {product.color}
                  </div>
                </div>
              </div>
              <div className={'product-price-info-wrapper'}>
                <div className={'product-price-element'}>
                  {product.discountedPrice.toFixed(2)} TL
                </div>
                <div className={'product-old-price-element'}>
                  <div className={'product-old-price'}>
                    {product.price.toFixed(2)} TL
                  </div>
                  <div className={'product-discount'}>
                    {product.discount} %
                  </div>
                </div>
              </div>
            </div>
            <div className={'product-add-cart-wrapper'}>
              {
                GeneralHelper.hasProduct(basket, product.id) ? (
                  <button className={'product-button'} disabled={true}>
                    Bu ürünü sepete ekleyemezsiniz.
                  </button>
                ) : (
                  <button className={'product-button'} onClick={() => addProductToBasket(product.id)}>
                    Sepete Ekle
                  </button>
                )
              }
            </div>
          </div>
        ))
      )}
    </div>
  )
}