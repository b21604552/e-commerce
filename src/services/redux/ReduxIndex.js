import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
