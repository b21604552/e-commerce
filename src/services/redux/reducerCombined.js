import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
