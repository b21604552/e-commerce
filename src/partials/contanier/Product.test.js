import React from 'react'
import ReactDOM from 'react-dom'
import Product from './Product'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'

const data = [
  {
    'id': 0,
    'title': 'Apple iPhone 11 32GB',
    'brand': 'Apple',
    'color': 'Siyah',
    'price': 124,
    'discountedPrice': 109.12,
    'discount': 12,
    'image': 'Apple'
  },
  {
    'id': 1,
    'title': 'Apple iPhone 11 32GB',
    'brand': 'Apple',
    'color': 'Lacivert',
    'price': 124,
    'discountedPrice': 109.12,
    'discount': 12,
    'image': 'Apple'
  },
  {
    'id': 2,
    'title': 'Apple iPhone 11 32GB',
    'brand': 'Apple',
    'color': 'Sarı',
    'price': 124,
    'discountedPrice': 109.12,
    'discount': 12,
    'image': 'Apple'
  },
  {
    'id': 3,
    'title': 'Apple iPhone 11 32GB',
    'brand': 'Apple',
    'color': 'Beyaz',
    'price': 124,
    'discountedPrice': 109.12,
    'discount': 12,
    'image': 'Apple'
  },
  {
    'id': 4,
    'title': 'Apple iPhone 11 64GB',
    'brand': 'Apple',
    'color': 'Siyah',
    'price': 224,
    'discountedPrice': 194.88,
    'discount': 13,
    'image': 'Apple'
  },
  {
    'id': 5,
    'title': 'Apple iPhone 11 64GB',
    'brand': 'Apple',
    'color': 'Lacivert',
    'price': 224,
    'discountedPrice': 194.88,
    'discount': 13,
    'image': 'Apple'
  },
  {
    'id': 6,
    'title': 'Apple iPhone 11 64GB',
    'brand': 'Apple',
    'color': 'Sarı',
    'price': 224,
    'discountedPrice': 194.88,
    'discount': 13,
    'image': 'Apple'
  },
  {
    'id': 7,
    'title': 'Apple iPhone 11 64GB',
    'brand': 'Apple',
    'color': 'Beyaz',
    'price': 224,
    'discountedPrice': 194.88,
    'discount': 13,
    'image': 'Apple'
  },
  {
    'id': 8,
    'title': 'Apple iPhone 11 128GB',
    'brand': 'Apple',
    'color': 'Siyah',
    'price': 324,
    'discountedPrice': 278.64,
    'discount': 14,
    'image': 'Apple'
  },
  {
    'id': 9,
    'title': 'Apple iPhone 11 128GB',
    'brand': 'Apple',
    'color': 'Lacivert',
    'price': 324,
    'discountedPrice': 278.64,
    'discount': 14,
    'image': 'Apple'
  },
  {
    'id': 10,
    'title': 'Apple iPhone 11 128GB',
    'brand': 'Apple',
    'color': 'Sarı',
    'price': 324,
    'discountedPrice': 278.64,
    'discount': 14,
    'image': 'Apple'
  },
  {
    'id': 11,
    'title': 'Apple iPhone 11 128GB',
    'brand': 'Apple',
    'color': 'Beyaz',
    'price': 324,
    'discountedPrice': 278.64,
    'discount': 14,
    'image': 'Apple'
  }
]

describe('Product Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <Product data={data}/>
    </Provider>
  )

  it('Renders Product without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders Product correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })
})