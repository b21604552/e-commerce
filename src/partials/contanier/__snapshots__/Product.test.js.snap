// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Test Renders Product correctly 1`] = `
<div
  className="products-wrapper"
>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-0"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 32GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Siyah
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          109.12
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            124.00
             TL
          </div>
          <div
            className="product-discount"
          >
            12
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-1"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 32GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Lacivert
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          109.12
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            124.00
             TL
          </div>
          <div
            className="product-discount"
          >
            12
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-2"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 32GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            SarÄ±
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          109.12
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            124.00
             TL
          </div>
          <div
            className="product-discount"
          >
            12
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-3"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 32GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Beyaz
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          109.12
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            124.00
             TL
          </div>
          <div
            className="product-discount"
          >
            12
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-4"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 64GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Siyah
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          194.88
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            224.00
             TL
          </div>
          <div
            className="product-discount"
          >
            13
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-5"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 64GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Lacivert
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          194.88
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            224.00
             TL
          </div>
          <div
            className="product-discount"
          >
            13
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-6"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 64GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            SarÄ±
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          194.88
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            224.00
             TL
          </div>
          <div
            className="product-discount"
          >
            13
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-7"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 64GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Beyaz
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          194.88
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            224.00
             TL
          </div>
          <div
            className="product-discount"
          >
            13
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-8"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 128GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Siyah
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          278.64
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            324.00
             TL
          </div>
          <div
            className="product-discount"
          >
            14
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-9"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 128GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Lacivert
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          278.64
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            324.00
             TL
          </div>
          <div
            className="product-discount"
          >
            14
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-10"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 128GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            SarÄ±
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          278.64
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            324.00
             TL
          </div>
          <div
            className="product-discount"
          >
            14
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
  <div
    className="product-wrapper"
  >
    <div
      className="product-image-wrapper"
    >
      <img
        alt="product-11"
        src="Apple.png"
      />
    </div>
    <div
      className="product-info-wrapper"
    >
      <div
        className="product-title"
      >
        Apple iPhone 11 128GB
      </div>
      <div
        className="product-properties-info-wrapper"
      >
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Marka:
          </div>
          <div
            className="product-properties-value"
          >
            Apple
          </div>
        </div>
        <div
          className="product-properties-element"
        >
          <div
            className="product-properties-label"
          >
            Renk:
          </div>
          <div
            className="product-properties-value"
          >
            Beyaz
          </div>
        </div>
      </div>
      <div
        className="product-price-info-wrapper"
      >
        <div
          className="product-price-element"
        >
          278.64
           TL
        </div>
        <div
          className="product-old-price-element"
        >
          <div
            className="product-old-price"
          >
            324.00
             TL
          </div>
          <div
            className="product-discount"
          >
            14
             %
          </div>
        </div>
      </div>
    </div>
    <div
      className="product-add-cart-wrapper"
    >
      <button
        className="product-button"
        onClick={[Function]}
      >
        Sepete Ekle
      </button>
    </div>
  </div>
</div>
`;
